### PR TITLE
Added entity_id to history graph tooltip

### PR DESCRIPTION
**Pull request by tobias-kuendig**
## Proposed change

For this year's Hacktoberfest I wanted to finally deep-dive into my favourite open-source project of this year! This is just a minor addition but something that has irked me for a few weeks now.

This PR adds the entity_id as a `beforeBody` text to the timeline history graph tooltip. 

ha

If there are entities with the same name (like a group + light or a device_tracker + person), this change makes it easy to identify what data you are looking at.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New (minor!) feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

This PR makes the Chart.js `beforeBody` property usable in chart tooltips. The property is documented here: https://www.chartjs.org/docs/latest/configuration/tooltip.html

### CODE REVIEW

**Comment by homeassistant** [comment]
Hi @tobias-kuendig, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;frontend#7310).

Once you do that we will be able to review and accept this pull request.

Thanks!

**Comment by bramkragten** [comment]
We should only show the `entity_id` when in advanced mode

**Comment by tobias-kuendig** [comment]
Good point, I have update this PR.

File: src/components/state-history-chart-timeline.js [hunk 1]
```diff
@@ -174,7 +174,7 @@ class StateHistoryChartTimeline extends LocalizeMixin(PolymerElement) {
     };
 
     const formatTooltipBeforeBody = (item, data) => {
-      if (!item[0]) {
+      if (!this.hass.userData?.showAdvanced || !item[0]) {
```
Comment (bramkragten) [inline src/components/state-history-chart-timeline.js:177]: This is a javascript file, not typescript. So you can't use optional chaining.

Comment (bramkragten) [inline src/components/state-history-chart-timeline.js:177]:
```suggestion
      if (!this.hass.userData || !this.hass.userData.showAdvanced || !item[0]) {
```

Comment (tobias-kuendig) [reply src/components/state-history-chart-timeline.js]: Sorry, that was a copy paste error. I have applied your suggestion.

**Review by bramkragten — APPROVED** [review]

**Pull request merged**